tests/test_weighted_pandas.py

```diff
@@ -76,124 +84,220 @@ def test_WeightedDataFrame_key(frame):
     assert isinstance(mean, WeightedSeries)
+    assert_array_equal(mean, frame.T.mean())
+    assert isinstance(frame.T.mean(), WeightedSeries)
+    assert mean._weighted
+    assert frame.T.mean()._weighted
+    assert_array_equal(mean.index, frame.index)
     assert_allclose(mean.mean(), 0.5, atol=1e-2)
 
 
 def test_WeightedDataFrame_std(frame):
     std = frame.std()
-    assert isinstance(std, Series)
+    assert isinstance(std, WeightedSeries)
+    assert not std._weighted
+    assert_array_equal(std.index, frame.columns)
     assert_allclose(std, (1./12)**0.5, atol=1e-2)
 
     std = frame.std(axis=1)
     assert isinstance(std, WeightedSeries)
+    assert_array_equal(std, frame.T.std())
+    assert isinstance(frame.T.std(), WeightedSeries)
+    assert std._weighted
+    assert frame.T.std()._weighted
+    assert_array_equal(std.index, frame.index)
     assert_allclose(std.mean(), (1./12)**0.5, atol=1e-1)
 
 
 def test_WeightedDataFrame_cov(frame):
     cov = frame.cov()
-    assert isinstance(cov, DataFrame)
-    assert_allclose(cov, (1./12)*np.identity(3), atol=1e-2)
+    assert isinstance(cov, WeightedDataFrame)
+    assert not cov._weighted(0) and not cov._weighted(1)
+    assert_array_equal(cov.index, frame.columns)
+    assert_array_equal(cov.columns, frame.columns)
+    assert_allclose(cov, (1./12)*np.identity(6), atol=1e-2)
+
+    cov = frame[:5].T.cov()
+    assert isinstance(cov, WeightedDataFrame)
+    assert cov._weighted(0) and cov._weighted(1)
+    assert_array_equal(cov.index, frame[:5].index)
+    assert_array_equal(cov.columns, frame[:5].index)
 
 
 def test_WeightedDataFrame_corr(frame):
     corr = frame.corr()
-    assert isinstance(corr, DataFrame)
-    assert_allclose(corr, np.identity(3), atol=1e-2)
+    assert isinstance(corr, WeightedDataFrame)
+    assert not corr._weighted(0) and not corr._weighted(1)
+    assert_array_equal(corr.index, frame.columns)
+    assert_array_equal(corr.columns, frame.columns)
+    assert_allclose(corr, np.identity(6), atol=1.1e-2)
+
+    corr = frame[:5].T.corr()
+    assert isinstance(corr, WeightedDataFrame)
+    assert corr._weighted(0) and corr._weighted(1)
+    assert_array_equal(corr.index, frame[:5].index)
+    assert_array_equal(corr.columns, frame[:5].index)
 
 
 def test_WeightedDataFrame_corrwith(frame):
     correl = frame.corrwith(frame.A)
-    assert isinstance(correl, Series)
-    assert_allclose(frame.corrwith(frame.A), frame.corr()['A'])
+    assert isinstance(correl, WeightedSeries)
+    assert not correl._weighted
+    assert_array_equal(correl.index, frame.columns)
+    assert_allclose(correl, frame.corr()['A'])
 
     correl = frame.corrwith(frame[['A', 'B']])
-    assert isinstance(correl, Series)
+    assert isinstance(correl, WeightedSeries)
+    assert not correl._weighted
+    assert_array_equal(correl.index, frame.columns)
     assert_allclose(correl['A'], 1, atol=1e-2)
     assert_allclose(correl['B'], 1, atol=1e-2)
     assert np.isnan(correl['C'])
```

Comment:
What about a test case where for `frame.corrwith(other)`:
* `frame` is unweighted?
* `other` is unweighted?

Reply:
At the moment this fails with `ValueError: cannot join with no overlapping index names`. I think that this is the correct behaviour, since it forbids correlating weighted and unweighted series (when strictly any combination (addition, multiplication, correlation) of series for weighted operations should have the same weights).

Reply:
I've added a test that checks for this failure.